sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/test/TestTableProvider.java

```diff
@@ -90,7 +94,38 @@ public void dropTable(String tableName) {
 
   @Override
   public synchronized BeamSqlTable buildBeamSqlTable(Table table) {
-    return new InMemoryTable(tables().get(table.getName()));
+    BeamSqlTable beamSqlTable;
+
+    if (table.getProperties().containsKey(PUSH_DOWN)) {
```

Comment:
It seems like table should have a property with if it supports PUSH_DOWN or not? Why it can not be a filterable interface detection and has a BeamIOFilterableSource?

Reply:
The reason I decided to make it configurable with a property is to not break existing code while working on push-down. I have updated the code to utilize the existing `InMemoryTable` class. I am not sure if interface detection would would still be applicable here, could you please take another look at the updated file?